pkg/virt-controller/network/multus_annotations.go

```diff
@@ -56,18 +56,23 @@ func (mnap multusNetworkAnnotationPool) toString() (string, error) {
 	return string(multusNetworksAnnotation), nil
 }
 
-func GenerateMultusCNIAnnotation(namespace string, interfaces []v1.Interface, networks []v1.Network, config *virtconfig.ClusterConfig) (string, error) {
-	return GenerateMultusCNIAnnotationFromNameScheme(namespace, interfaces, networks, namescheme.CreateHashedNetworkNameScheme(networks), config)
+func GenerateMultusCNIAnnotation(namespace string, interfaces []v1.Interface, networks []v1.Network, networkToIPAMClaimParams map[string]IPAMClaimParams, config *virtconfig.ClusterConfig) (string, error) {
+	return GenerateMultusCNIAnnotationFromNameScheme(namespace, interfaces, networks, namescheme.CreateHashedNetworkNameScheme(networks), networkToIPAMClaimParams, config)
 }
 
-func GenerateMultusCNIAnnotationFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, config *virtconfig.ClusterConfig) (string, error) {
+func GenerateMultusCNIAnnotationFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, networkToIPAMClaimParams map[string]IPAMClaimParams, config *virtconfig.ClusterConfig) (string, error) {
 	multusNetworkAnnotationPool := multusNetworkAnnotationPool{}
 
+	persistentIPsEnabled := config.PersistentIPsEnabled()
```

Comment:
isn't it a little bit too late to be checking this ? 
IMHO (or at least the way I pictured this) was for this check to be made on the client, and if the feature gate is not enabled, you'd log it / throw event (not sure if the latter actually makes sense) and pass along an empty map of ipam claims. Point being it does not make sense to throw an error and interrupt the whole flow.

Reply:
Right, done